debian/Dockerfile

```diff
@@ -24,8 +24,23 @@ RUN cd && wget https://nixos.org/releases/nix/nix-$NIX_VERSION/nix-$NIX_VERSION-
     && ~/nix-*-x86_64-linux/install \
     && rm -rf ~/nix-*-*
 
+
+# < Nix store as a volume
+# We want to be able to define /nix/store as a volume
+# We thus need to "save" the current nix store to be able
+# to restore it at startup time
+RUN cp -R /nix/store /nix/initial-store
+VOLUME ["/nix/store"]
+# />
+
 ENV ENV="/home/nixuser/.nix-profile/etc/profile.d/nix.sh"
-RUN echo ". ${ENV}" >> ${HOME}/.profile
+# < Create bash profile
+# Copy "saved" store if the /nix/store is empty
+RUN echo "[ ! \"$(ls -A /nix/store)\" ] && cp -R /nix/initial-store/* /nix/store/" >> ${HOME}/.profile
```

Comment:
this will run every time the image is used in a container?

Reply:
It will run everytime this container starts yes.

(:memo: A container image is not really "used" in a container, it's more that a container uses a container image)

Reply:
Couldn't we more rsync to avoid copying if already present ?

Reply:
(nevermind, i somehow ignored the `echo`)

Reply:
Hm, this enforces us to install rsync on the base docker nix image. I don't really feel like it as this is a public nix image we publish under our name.

Isn't the bash check `[ ! "$(ls -A /nix/store)" ]` enough @gaetanfl?

Reply:
Didn't see the check, sorry. It's fine like this